pkg/volume/csi/csi_attacher.go

```diff
@@ -219,37 +220,39 @@ func (c *csiAttacher) GetDeviceMountPath(spec *volume.Spec) (string, error) {
 	return deviceMountPath, nil
 }
 
-func (c *csiAttacher) MountDevice(spec *volume.Spec, devicePath string, deviceMountPath string) (err error) {
+func (c *csiAttacher) MountDevice(spec *volume.Spec, devicePath string, deviceMountPath string) (volumetypes.OperationStatus, error) {
 	klog.V(4).Infof(log("attacher.MountDevice(%s, %s)", devicePath, deviceMountPath))
+	// lets default to operation as finished state
+	opExitStatus := volumetypes.OperationFinished
 
 	if deviceMountPath == "" {
-		return errors.New(log("attacher.MountDevice failed, deviceMountPath is empty"))
+		return opExitStatus, errors.New(log("attacher.MountDevice failed, deviceMountPath is empty"))
 	}
 
 	mounted, err := isDirMounted(c.plugin, deviceMountPath)
 	if err != nil {
 		klog.Error(log("attacher.MountDevice failed while checking mount status for dir [%s]", deviceMountPath))
-		return err
+		return opExitStatus, err
```

Comment:
is there some possible state that temporarily fail  to get mount state? so operationstatus is nochange?

Reply:
I am not sure how this could happen and if this does - since we have not started `NodeStageVolume` call yet, it should be safe to mark the operation as "finished".  Are you thinking some corner case where `NodeStageVolume` was called but is in-progress and next `MountDevice` call fails in `isDirMounted` check and hence device is marked as not-mounted?

Reply:
yes, that is the thought